tests/test_encodings.py

```diff
@@ -0,0 +1,383 @@
+# Copyright 2022 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+from typing import Any
+
+import numpy as np
+import pytest
+from PIL import Image
+
+import streaming.base.format.json.encodings as jsonEnc
+import streaming.base.format.mds.encodings as mdsEnc
+import streaming.base.format.xsv.encodings as xsvEnc
+
+
+class TestMDSEncodings:
+
+    @pytest.mark.parametrize('data', [b'5', b'\x00\x00'])
+    def test_byte_encode_decode(self, data: bytes):
+        byte_enc = mdsEnc.Bytes()
+        assert byte_enc.size is None
+        output = byte_enc.encode(data)
+        assert output == data
+        output = byte_enc.decode(data)
+        assert output == data
+
+    @pytest.mark.parametrize('data', ['9', 25])
+    def test_byte_encode_invalid_data(self, data: Any):
+        with pytest.raises(AttributeError):
+            byte_enc = mdsEnc.Bytes()
+            _ = byte_enc.encode(data)
+
+    @pytest.mark.parametrize(('data', 'encode_data'),
+                             [('99', b'99'), ('streaming dataset', b'streaming dataset')])
+    def test_str_encode_decode(self, data: str, encode_data: bytes):
+        str_enc = mdsEnc.Str()
+        assert str_enc.size is None
+
+        # Test encode
+        enc_data = str_enc.encode(data)
+        assert isinstance(enc_data, bytes)
+        assert enc_data == encode_data
+
+        # Test decode
+        dec_data = str_enc.decode(encode_data)
+        assert isinstance(dec_data, str)
+        assert dec_data == data
+
+    @pytest.mark.parametrize('data', [b'9', 25])
+    def test_str_encode_invalid_data(self, data: Any):
+        with pytest.raises(AttributeError):
+            str_enc = mdsEnc.Str()
+            _ = str_enc.encode(data)
+
+    @pytest.mark.parametrize(('data', 'encode_data'), [(99, b'c\x00\x00\x00\x00\x00\x00\x00'),
+                                                       (987654321, b'\xb1h\xde:\x00\x00\x00\x00')])
+    def test_int_encode_decode(self, data: int, encode_data: bytes):
+        int_enc = mdsEnc.Int()
+        assert int_enc.size == 8
+
+        # Test encode
+        enc_data = int_enc.encode(data)
+        assert isinstance(enc_data, bytes)
+        assert enc_data == encode_data
+
+        # Test decode
+        dec_data = int_enc.decode(encode_data)
+        assert isinstance(dec_data, int)
+        assert dec_data == data
+
+    @pytest.mark.parametrize('data', [b'9', 25.9])
+    def test_int_encode_invalid_data(self, data: Any):
+        with pytest.raises(AttributeError):
+            int_enc = mdsEnc.Int()
+            _ = int_enc.encode(data)
+
+    @pytest.mark.parametrize('mode', ['I', 'L', 'RGB'])
+    def test_pil_encode_decode(self, mode: str):
+        pil_enc = mdsEnc.PIL()
+        assert pil_enc.size is None
+
+        # Creating the 32 X 32 NumPy Array with random values
```

Comment:
maybe conventionally a lowercase x here and elsewhere?

Reply:
Done in https://github.com/mosaicml/streaming/pull/72/commits/f202a4ff6281c757fc5cc6f0d11e3e2edfbef424.